mlflow/server/js/src/gateway/components/endpoints/EndpointsList.tsx

```diff
@@ -0,0 +1,343 @@
+  Input,
+  LinkIcon,
+  PencilIcon,
+  SearchIcon,
+  Spinner,
+  Table,
+  TableCell,
+  TableHeader,
+  TableRow,
+  Tag,
+  Tooltip,
+  TrashIcon,
+  Typography,
+  useDesignSystemTheme,
+} from '@databricks/design-system';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { Link } from '../../../common/utils/RoutingUtils';
+import { useEndpointsListData } from '../../hooks/useEndpointsListData';
+import { formatProviderName } from '../../utils/providerUtils';
+import { timestampToDate } from '../../utils/dateUtils';
+import { TimeAgo } from '../../../shared/web-shared/browse/TimeAgo';
+import { EndpointsFilterButton, type EndpointsFilter } from './EndpointsFilterButton';
+import { EndpointBindingsDrawer } from './EndpointBindingsDrawer';
+import { DeleteEndpointModal } from './DeleteEndpointModal';
+import GatewayRoutes from '../../routes';
+import type { Endpoint, EndpointBinding } from '../../types';
+
+interface EndpointsListProps {
+  onEndpointDeleted?: () => void;
+}
+
+export const EndpointsList = ({ onEndpointDeleted }: EndpointsListProps) => {
+  const { theme } = useDesignSystemTheme();
+  const { formatMessage } = useIntl();
+  const [searchFilter, setSearchFilter] = useState('');
+  const [filter, setFilter] = useState<EndpointsFilter>({ providers: [] });
+  const [bindingsDrawerEndpoint, setBindingsDrawerEndpoint] = useState<{
+    endpointId: string;
+    endpointName: string;
+    bindings: EndpointBinding[];
+  } | null>(null);
+  const [deleteModalEndpoint, setDeleteModalEndpoint] = useState<Endpoint | null>(null);
+
+  const { endpoints, filteredEndpoints, isLoading, availableProviders, getBindingsForEndpoint, refetch } =
+    useEndpointsListData({ searchFilter, filter });
+
+  const handleDeleteClick = (endpoint: Endpoint) => {
+    setDeleteModalEndpoint(endpoint);
+  };
+
+  const handleDeleteSuccess = () => {
+    setDeleteModalEndpoint(null);
+    refetch();
+    onEndpointDeleted?.();
+  };
+
+  if (isLoading || !endpoints) {
+    return (
+      <div
+        css={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+          gap: theme.spacing.sm,
+          padding: theme.spacing.lg,
+          minHeight: 200,
+        }}
+      >
+        <Spinner size="small" />
+        <FormattedMessage defaultMessage="Loading endpoints..." description="Loading message for endpoints list" />
+      </div>
+    );
+  }
+
+  const isFiltered = searchFilter.trim().length > 0 || filter.providers.length > 0;
+
+  const getEmptyState = () => {
+    const isEmptyList = endpoints?.length === 0;
+    if (filteredEndpoints.length === 0 && isFiltered) {
+      return (
+        <Empty
+          title={
+            <FormattedMessage
+              defaultMessage="No endpoints found"
+              description="Empty state title when filter returns no results"
+            />
+          }
+          description={null}
+        />
+      );
+    }
+    if (isEmptyList) {
```

Comment:
nit: do we need this check since empty state is displayed only when there's no row?